File: ios/App/Shared/Core/Date/AppDate.swift
//
// Created by Shaban on 22/06/2023.
//

import Foundation

enum DateFormat: String {
    case yyyyMMdd = "yyyy-MM-dd"
}

class AppDate {
    static func format(_ date: Date, dateFormat: DateFormat = .yyyyMMdd) -> String {
        let formatter = DateFormatter()
        formatter.dateFormat = dateFormat.rawValue
        return formatter.string(from: date)
    }

    static func date(from string: String, withFormat format: DateFormat = .yyyyMMdd) -> Date {
        let formatter = DateFormatter()
        formatter.dateFormat = format.rawValue
        return formatter.date(from: string) ?? Date()
    }
}

File: flutter_module/lib/shared/core/date/app_date.dart
import 'package:intl/intl.dart';

enum AppDateFormat {
  yyyyMMdd,
}

extension AppDateFormatExtension on AppDateFormat {
  String get formatString {
    switch (this) {
      case AppDateFormat.yyyyMMdd:
        return 'yyyy-MM-dd';
      default:
        return 'yyyy-MM-dd';
    }
  }
}

class AppDate {
  static String format(DateTime date,
      {AppDateFormat format = AppDateFormat.yyyyMMdd}) {
    final formatter = DateFormat(format.formatString);
    return formatter.format(date);
  }
}
